samcli/lib/utils/managed_cloudformation_stack.py

```diff
@@ -0,0 +1,130 @@
+"""
+Bootstrap's user's development environment by creating cloud resources required by SAM CLI
+"""
+
+import logging
+
+import boto3
+
+import click
+
+from botocore.config import Config
+from botocore.exceptions import ClientError, BotoCoreError, NoRegionError, NoCredentialsError, ProfileNotFound
+
+from samcli.commands.bootstrap.exceptions import ManagedStackError
+from samcli.commands.exceptions import UserException, CredentialsError, RegionError
+
+
+SAM_CLI_STACK_PREFIX = "aws-sam-cli-managed-"
+LOG = logging.getLogger(__name__)
+
+
+def manage_stack(profile, region, stack_name, template_body):
+    try:
+        if profile:
+            session = boto3.Session(profile_name=profile, region_name=region if region else None)
+            cloudformation_client = session.client("cloudformation")
+        else:
+            cloudformation_client = boto3.client(
+                "cloudformation", config=Config(region_name=region if region else None)
+            )
+    except ProfileNotFound as ex:
+        raise CredentialsError(
+            f"Error Setting Up Managed Stack Client: the provided AWS name profile '{profile}' is not found. "
+            "please check the documentation for setting up a named profile: "
+            "https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html"
+        ) from ex
```

Comment:
this is a new logic but doesn't not result in any behavior change as the method caller(see bootstrap.py) always pass profile=None (which how it used to be) so this `ProfileNotFound`exception will never be raised